LiteFarmOrg/LiteFarm
# LF-2938: When assigning a user to a task during task creation flow, prompt for wage as needed

**Description**

Show additional inputs on the task assignment view for task creation so that the user can set worker's wage or a wage for a task.

- update `useTaskAssignForm` hook to take `defaultAssignee` and `disableUnAssignedOption` and return more functions to make it reusable
- add `<AssignTask />` to `<PureTaskAssignment />`
- call `useTaskAssignForm` in `<TaskAssignement />`
- move some code from `<PureTaskAssignment />` up to `<TaskAssignement />`
- make the override wage input required (I removed "optional" tag, but the functionality was missing...)
- update the max number accepted for the override wage input to make it the same as the other input

Jira link: https://lite-farm.atlassian.net/browse/LF-2938

**Type of change**


**How Has This Been Tested?**

- go to `/tasks`, click "Create a task" and proceed to task assignment(`/add_task/task_assignment`)
   - if you don't see other users in the assignee dropdown, you should be selected as default assignee
   - **[farm worker]** you only see the assignee select, and you and "Unassigned" is the only options for the select
   - **[admin]** if there are multiple workers, "Unassigned" option should be selected as default
     - select "Unassigned", and override:
        - yes -> input wage -> save => task is created with wage_at_moment without assignee
        - no -> save -> task is created without  `wage_at_moment` and assignee
     - select a worker with a wage or a worker who  has “don’t ask again” set to true
        - you only see "Do you need to override the assignees wage for this task?" and the radio buttons.
     - select a worker without a wage
       - yes, set hourly wage -> save => a task is created with the assignee `without wage_at_moment`. User's wage is updated in the `userFarm` table
       - yes, just for this task -> save => a task is created with the assignee and `wage_at_moment`
       - no -> save => a task is created with assignee without `wage_at_moment`
       - Don't ask again -> save => a task is created with assignee without `wage_at_moment`. User's `do_not_ask_again` is updated in the `userFarm` table


**Checklist:**

- [ ] I have added the GNU General Public License shown below to all new files
